src/server/server_family.cc

```diff
@@ -2872,11 +2873,8 @@ void ServerFamily::ReplicaOfInternal(CmdArgList args, Transaction* tx, SinkReply
     builder->SendError(ec ? ec.Format() : "replication cancelled");
     return;
   }
-  // Successfully connected now we flush
-  // If we are called by "Replicate", tx will be null but we do not need
-  // to flush anything.
+
   if (on_err == ActionOnConnectionFail::kReturnOnError) {
-    Drakarys(tx, DbSlice::kDbAll);
```

Comment:
Currently the flush in the replication fiber is done after some protocol commands run in the InitiatePSync/InitiateDflySync. But if this fails we will not flush the database, the replica will continue to reconnect untill replicaof no one will be executed.
I am not sure about this change as we have some data in the server although replication was not done and this can lead to mistakes as the user might thing that some data was replicated and decide to move to the replica as a master although the data there is not relevant. 
So I do think it makes more sense to move this flush inside the replicaiton code but the flush should be done first thing I believe before we start some protocol commands that might return error

Reply:
> But if this fails we will not flush the database, the replica will continue to reconnect untill replicaof no one will be executed. 

Won't we restart dfly/psync for that case which will eventually call and flush the datastore anyway ? If we have an error or a reconnect, we still need to restart full sync before we enter stable sync

Reply:
@adiholden  on the other hand it means we loose data before even we know we have a chance to replicate. I think I opened an issue that asks not to destroy the data on a server if `replicaof ` is going to fail during the initial handshake. How come it was closed if we flush before executing the replicaof transaction.

Reply:
If the initial handshake failes - the replica::start() flow we will return error to the client and will not flush.
But if we do return ok to client and later in this process we fail to replicate does it makes sense that the data in the replica will not be result of replicaiton?

Reply:
no, I think it should be consistent and if we returned OK, should either succeed or be empty.